src/emscripten-optimizer/istring.h

```diff
@@ -127,6 +127,8 @@ struct IString {
     return !str || str[0] == 0;
   }
 
+  operator bool() const { return !!*this; }
```

Comment:
Shouldn't this check `str`? Or am I missing something?

If I'm not missing something then I'm fine with an API like this. If so then we should be consistent, though, and remove `is()` and `isNull()`.

Reply:
This uses `operator!` from above, which checks `str`. So this returns `true` iff the string exists and is not empty.

Reply:
Oh, I see...

Ok, sounds good, assuming you are on board with the followup plan?

Reply:
I'm a little concerned about replacing `is` and `isNull` because they're not technically the same as this `operator bool`. Does the distinction between an empty string and a null string ever matter in practice? Should I be using `.is()` in this PR rather than treating null strings the same as empty strings?

Reply:
@kripken what do you think about this? Does the difference between this `operator bool` and `is` matter?

Reply:
I don't think the difference matters, but if it does then the test suite should catch it.